Sourcepawn new syntax

TheChineese opened the pull request:
Updated to use sourcemod's new syntax (introduced by version 1.7). Especially new type for ConVars (ConVar). So that the ConVar Methodmap can be used (i.e. g_cTest.GetString(buffer, 4)).

TheChineese committed:
Sourcepawn new syntax

Updated to use sourcemod's new syntax (introduced by version 1.7). Especially new type for ConVars (ConVar).

Impact123 commented:
There are a few things missing like using `null` instead of `INVALID_HANDLE` and using the methodmap syntax to close handles of files, but other than that it looks good. Thanks!
The reason i haven't done this before is because i wanted to support users who, for whatever reason,  still use an old compiler. [I've used the **view_as** operator in CallAdmin to be able to use methodmaps](https://github.com/Impact123/CallAdmin/blob/46036f4ecd7fedd095cbaa30747a486de6940685/gameserver/calladmin.sp#L324), But now that i think about this, there's no reason to not update the syntax.

TheChineese committed:
Fixes

Changed return type of AutoExecConfig_SetCreateFile; Changed
INVALID_HANDLE to null. Now using new File type.

TheChineese commented:
Changed the mentioned points. Did I forgot something?

Bara20 commented:
btw. you can use `#pragma newdecls required` for testing.

Impact123 commented:
This looks good, thanks for the effort!

TheChineese commented:
@Bara20 Did that, but there was no error while using the old syntax for the file handling nor for the decleration of INVALID_HANDLE.
